src/Server/Provision/ServerProvisioner.php

```diff
@@ -163,26 +163,35 @@ public function check()
      */
     private function checkEntities()
     {
-        if ($entities = $this->getEntities()) {
-            return $entities;
+        $isCheck = true;
+        if ($entities = $this->getEntities($isCheck)) {
+            return $entities[0];
         }
 
         throw new \Exception('No matching Entity found in inventory');
     }
 
     /**
-     * @return Entity|void
+     * @param boolean $isCheck
+     *
+     * @return array|void
```

Comment:
I don't think it's possibly for this to return null/void any more, since we're always returning an array. Also we can be more specific about the type of array since it's an array of Entities.
```suggestion
     * @return Entity[]
```